scout/adapter/mongo/variant_events.py

```diff
@@ -246,6 +246,44 @@ def validated(self, institute_id=None, user_id=None, case_id=None):
         sanger_ordered = [item for item in results]
         return sanger_ordered
 
+    def matching_tiered(self, query_variant, user_institutes):
+        """Retrieve all tiered tags assigned to a cancer variant in other cases (accessible to the user)
+        Args:
+            query_variant(dict)
+            user_institutes(list): list of dictionaries
+
+        Returns:
+            matching_tier(set)  # set of tuples like this: (tier, label_class, link_to_variant)
+        """
+        tiered = set()
+        query = {
+            "category": "variant",
+            "verb": "cancer_tier",
+            "subject": query_variant["display_name"],
+            "institute": {"$in": [inst["_id"] for inst in user_institutes]},
+        }
+
+        for tiered_event in self.event_collection.find(query):
+            # Check if variant is still tiered as suggested by the event
+            tiered_matching_variant = self.variant(
+                case_id=tiered_event["case"], document_id=tiered_event["variant_id"]
+            )
+            if (
+                tiered_matching_variant is None
+                or tiered_matching_variant["_id"] == query_variant["_id"]
+            ):
+                continue
```

Comment:
The loophole being if both a somatic and a constitutional variant have been tiered in the same case, but that is a rather odd situation and could potentially be useful. It does allow the list to grow fairly large if it is a common recurrent variant, but that is better handled on output I guess.

Reply:
So, 💯, after some consideration!